src/app/components/sign-up-modal/sign-up-modal.component.ts

```diff
@@ -104,84 +109,47 @@ openModal(template :TemplateRef<any>){
  * @memberof SignupModalComponent
  */
 submitUser() {
-    this.user.userId = 0;
-    this.firstNameError = '';
-    this.lastNameError = '';
-    this.phoneNumberError ='';
-    this.userNameError ='';
-    this.emailError ='';
-    this.hStateError='';
-    this.hAddressError='';
-    this.hCityError='';
-    this.hZipError='';
-    this.success='';
-    this.user.wAddress = this.user.hAddress;
-    this.user.wState = this.user.hState;
-    this.user.wCity = this.user.hCity;
-    this.user.wZip = this.user.hZip;
-    let driver = <HTMLInputElement> document.getElementById("driver");
-    let rider = <HTMLInputElement> document.getElementById("rider");
+  this.firstNameError = '';
+  this.lastNameError = '';
+  this.phoneNumberError ='';
+  this.userNameError ='';
+  this.emailError ='';
+  this.hStateError='';
+  this.hAddressError='';
+  this.hCityError='';
+  this.hZipError='';
+  this.success='';
+  this.user.car = null;
+  
+  let driver = <HTMLInputElement> document.getElementById("driver");
+  let rider = <HTMLInputElement> document.getElementById("rider");
 
-    if(driver.checked == true){
-      this.user.isDriver =  true;
-    }
-    if(rider.checked == true){
-      this.user.isDriver =  false;
-    }
-    this.userService.addUser(this.user).subscribe(
-      res => {
-        console.log(res);
-        let i = 0;
-        if(res.firstName != undefined){
-          this.firstNameError = res.firstName[0];
-          i = 1;
-        }
-        if(res.lastName != undefined){
-          this.lastNameError = res.lastName[0];
-          i = 1;
-
-        }
-        if(res.phoneNumber != undefined){
-          this.phoneNumberError = res.phoneNumber[0];
-          i = 1;
-
-        }
-        if(res.email != undefined){
-          this.emailError = res.email[0];
-          i = 1;
-
-        }
-        if(res.userName != undefined){
-          this.userNameError = res.userName[0];
-          i = 1;
-
-        }
-        if(res.hState != undefined){
-          this.hStateError = res.hState[0];
-          i = 1;
-
-        }
-        if(res.hAddress != undefined){
-          this.hAddressError = res.hAddress[0];
-          i = 1;
-
-        }
-        if(res.hCity != undefined){
-          this.hCityError = res.hCity[0];
-          i = 1;
-
-        }
-        if(res.hZip != undefined){
-          this.hZipError = res.hZip[0];
-          i = 1;
-
-        }
-        if(i === 0) {
-          i = 0;
-          this.success = "Registered successfully!";
-        }
+  if(driver.checked == true){
+    this.user.isDriver =  true;
+  }
+  if(rider.checked == true){
+    this.user.isDriver =  false;
+  }
+  this.userService.addUser(this.user).subscribe(
+    res => {
+      if (Object.keys(res).length === 0) {
+        this.http.get(`${environment.loginUri}?userName=${this.user.userName}&passWord='placeholder'`).subscribe(response => {
```

Comment:
This HTTP call should be moved to a service

Reply:
removed this request to authService